src/PostgREST/Response.hs

```diff
@@ -205,28 +209,32 @@ actionResponse (DbCallResult CallReadPlan{crMedia, crInvMthd=invMethod, crProc=p
         then Error.errorPayload $ Error.ApiRequestError $ Error.InvalidRange
           $ Error.OutOfBounds (show $ RangeQuery.rangeOffset iTopLevelRange) (maybe "0" show rsTableTotal)
         else LBS.fromStrict rsBody
+      isHeadMethod = invMethod == InvRead True
       prefHeader = maybeToList . prefAppliedHeader $ Preferences Nothing Nothing preferCount preferTransaction Nothing preferHandling preferTimezone preferMaxAffected []
+      cLHeader = if isHeadMethod then mempty else [contentLengthHeaderLazy rsOrErrBody]
       headers = contentRange : prefHeader
 
     let (status', headers', body) =
           if Routine.funcReturnsVoid proc then
               (HTTP.status204, headers, mempty)
             else
               (status,
-                headers ++ contentTypeHeaders crMedia ctxApiRequest,
-                if invMethod == InvRead True then mempty else rsOrErrBody)
+                headers ++ cLHeader ++ contentTypeHeaders crMedia ctxApiRequest,
+                if isHeadMethod then mempty else rsOrErrBody)
 
     (ovStatus, ovHeaders) <- overrideStatusHeaders rsGucStatus rsGucHeaders status' headers'
 
     Right $ PgrstResponse ovStatus ovHeaders body
 
   RSPlan plan ->
-    Right $ PgrstResponse HTTP.status200 (contentTypeHeaders crMedia ctxApiRequest) $ LBS.fromStrict plan
+    Right $ PgrstResponse HTTP.status200 (contentLengthHeader plan : contentTypeHeaders crMedia ctxApiRequest) $ LBS.fromStrict plan
 
 actionResponse (MaybeDbResult InspectPlan{ipHdrsOnly=headersOnly} body) _ versions conf sCache schema negotiatedByProfile =
-  Right $ PgrstResponse HTTP.status200
-    (MediaType.toContentType MTOpenAPI : maybeToList (profileHeader schema negotiatedByProfile))
-    (maybe mempty (\(x, y, z) -> if headersOnly then mempty else OpenAPI.encode versions conf sCache x y z) body)
+  let
+    rsBody = maybe mempty (\(x, y, z) -> if headersOnly then mempty else OpenAPI.encode versions conf sCache x y z) body
+    cLHeader = if headersOnly then mempty else [contentLengthHeaderLazy rsBody]
+  in
+  Right $ PgrstResponse HTTP.status200 (MediaType.toContentType MTOpenAPI : cLHeader ++ maybeToList (profileHeader schema negotiatedByProfile)) rsBody
```

Comment:
I can see that we include the header in openapi now

Reply:
Yes, it's now also included in RPC and errors too.